lib/api_auth/base.rb

```diff
@@ -78,23 +80,31 @@ def request_too_old?(headers)
       end
     end
 
-    def signatures_match?(headers, secret_key)
+    def signatures_match?(headers, secret_key, options)
+      options = options.merge(:with_http_method => true)
       if match_data = parse_auth_header(headers.authorization_header)
         hmac = match_data[2]
-        hmac == hmac_signature(headers, secret_key)
+        return (
```

Comment:
I think having parens where they aren't necessary can be confusing because people reading the code might assume they *are* necessary and that you're doing something kind of complicated here. At least that was my initial reaction to reading this. I think creating local variables might be good if you want this method to be extra clear.